miner/miner.go

```diff
@@ -0,0 +1,94 @@
+package miner
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"math"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	bc "github.com/ubclaunchpad/cumulus/blockchain"
+)
+
+const (
+	// MiningHeaderLen is the length of the MiningHeader struct in bytes
+	MiningHeaderLen = (3 * bc.HashLen) + (1 * (32 / 8)) + (1 * (64 / 8))
+)
+
+// MiningHeader contains the metadata required for mining
+type MiningHeader struct {
+	LastBlock bc.Hash
+	RootHash  bc.Hash
+	Target    bc.Hash
+	Time      uint32
+	Nonce     uint64
+}
+
+// Marshal converts a Mining Header to a byte slice
+func (mh *MiningHeader) Marshal() []byte {
+	var buf []byte
+	buf = append(buf, mh.LastBlock.Marshal()...)
+	buf = append(buf, mh.RootHash.Marshal()...)
+	buf = append(buf, mh.Target.Marshal()...)
+	tempBufTime := make([]byte, 4)
+	binary.LittleEndian.PutUint32(tempBufTime, mh.Time)
+	buf = append(buf, tempBufTime...)
+	tempBufNonce := make([]byte, 8)
+	binary.LittleEndian.PutUint64(tempBufNonce, mh.Nonce)
+	buf = append(buf, tempBufNonce...)
+	return buf
+}
+
+// SetMiningHeader sets the mining header (sets the time to the current time)
+func (mh *MiningHeader) SetMiningHeader(lastBlock bc.Hash, rootHash bc.Hash, target bc.Hash) {
+	mh.LastBlock = lastBlock
+	mh.RootHash = rootHash
+	mh.Target = target
+	mh.Time = uint32(time.Now().Unix())
+	mh.Nonce = 0
+}
+
+// VerifyProofOfWork computes the hash of the MiningHeader and returns true if the result is less than the target
+func (mh *MiningHeader) VerifyProofOfWork() bool {
+	if mh.DoubleHashSum().CompareTo(mh.Target, bc.LessThan) {
```

Comment:
Also here the namespacing does not really tell me what `LessThan` is. I wonder if 

+ `hash.GreaterThan(otherHash)`
+ `hash.LessThan(otherHash)`
+ `hash.EqualTo(otherHash)`

is a more clear API?